lib/bandit/clock.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Bandit.Clock do
+  @moduledoc false
+
+  use GenServer
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(_opts) do
+    __MODULE__ = :ets.new(__MODULE__, [:set, :protected, :named_table, {:read_concurrency, true}])
+    update_header()
+    timer_ref = Process.send_after(self(), :update, 1_000)
+    {:ok, %{timer_ref: timer_ref}}
```

Comment:
I think this would be better accomplished as a Task with a permanent restart strategy. No need for shutdown handling, no mailbox churn, just a simple tail recursion call.